plugin/src/test/java/org/opensearch/ml/rest/SecureMLRestIT.java

```diff
@@ -142,4 +160,151 @@ public void testTrainAndPredictWithFullAccess() throws IOException {
             }
         );
     }
+
+    public void testTrainModelWithFullAccessThenPredict() throws IOException {
+        KMeansParams kMeansParams = KMeansParams.builder().build();
+        // train model
+        train(mlFullAccessClient, FunctionName.KMEANS, irisIndex, kMeansParams, searchSourceBuilder, trainResult -> {
+            String modelId = (String) trainResult.get("model_id");
+            assertNotNull(modelId);
+            String status = (String) trainResult.get("status");
+            assertEquals(MLTaskState.COMPLETED.name(), status);
+            try {
+                getModel(mlFullAccessClient, modelId, model -> {
+                    String algorithm = (String) model.get("algorithm");
+                    assertEquals(FunctionName.KMEANS.name(), algorithm);
+                });
+            } catch (IOException e) {
+                assertNull(e);
+            }
+            try {
+                // predict with trained model
+                predict(mlFullAccessClient, FunctionName.KMEANS, modelId, irisIndex, kMeansParams, searchSourceBuilder, predictResult -> {
+                    String predictStatus = (String) predictResult.get("status");
+                    assertEquals(MLTaskState.COMPLETED.name(), predictStatus);
+                    Map<String, Object> predictionResult = (Map<String, Object>) predictResult.get("prediction_result");
+                    ArrayList rows = (ArrayList) predictionResult.get("rows");
+                    assertTrue(rows.size() > 1);
+                });
+            } catch (IOException e) {
+                assertNull(e);
+            }
+        }, false);
+    }
+
+    public void testTrainModelInAsyncWayWithFullAccess() throws IOException {
+        train(mlFullAccessClient, FunctionName.KMEANS, irisIndex, KMeansParams.builder().build(), searchSourceBuilder, trainResult -> {
+            assertFalse(trainResult.containsKey("model_id"));
+            String taskId = (String) trainResult.get("task_id");
+            assertNotNull(taskId);
+            String status = (String) trainResult.get("status");
+            assertEquals(MLTaskState.CREATED.name(), status);
+            try {
+                getTask(mlFullAccessClient, taskId, task -> {
+                    String algorithm = (String) task.get("function_name");
+                    assertEquals(FunctionName.KMEANS.name(), algorithm);
+                });
+            } catch (IOException e) {
```

Comment:
Is it possible that a different Exception be thrown here? Which would make the test fail.

Reply:
If the exception is not IOException, we expect the test to fail. So it's ok.